app/src/main/scala/org/alephium/explorer/persistence/queries/InputUpdateQueries.scala

```diff
@@ -0,0 +1,67 @@
+// Copyright 2018 The Alephium Authors
+// This file is part of the alephium project.
+//
+// The library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the library. If not, see <http://www.gnu.org/licenses/>.
+
+package org.alephium.explorer.persistence.queries
+
+import scala.concurrent.ExecutionContext
+
+import slick.dbio.DBIOAction
+import slick.jdbc.PostgresProfile.api._
+
+import org.alephium.explorer.api.model._
+import org.alephium.explorer.persistence._
+import org.alephium.explorer.persistence.schema.CustomGetResult._
+import org.alephium.util._
+
+object InputUpdateQueries {
+
+  def updateInputs()(implicit ec: ExecutionContext): DBActionWT[Int] = {
+    sql"""
+      UPDATE inputs
+      SET
+        output_ref_tx_hash = outputs.tx_hash,
+        output_ref_address = outputs.address,
+        output_ref_amount = outputs.amount
+      FROM outputs
+      WHERE inputs.output_ref_key = outputs.key
+      AND inputs.output_ref_tx_hash IS NULL
+      RETURNING outputs.address, inputs.tx_hash, inputs.block_hash, inputs.block_timestamp, inputs.tx_order, inputs.main_chain
```

Comment:
1. arg damn fork, mmmmm even if we do 
```
      WHERE inputs.output_ref_key = outputs.key
      AND inputs.output_ref_tx_hash IS NULL
      AND outputs.main_chain = true
```
it might be an issue if suddenly there's a reorg on the output-ref :/  

both `outputs.address` and `outputs.amount` are fine, as it's an immutable data. The problem is the `outputs.tx_hash`.... Thinking how we use it, it's only to create this little link in the UI: 
![image](https://user-images.githubusercontent.com/2979182/177156781-21b83ebf-6f6f-42a0-9e71-dcd293835141.png)
which redirect to the corresponding transaction. Maybe we could have a new endpoint: `/transaction-by-output-ref/<output_ref>` that we call when we click on that link. So we avoid a complicated mechanism to keep updated the the output ref tx hash in the input table.

Reply:
> Maybe we could have a new endpoint: /transaction-by-output-ref/<output_ref> that we call when we click on that link.

This sounds great to me 👍  We only update `outputRefAddress` and `outputRefAmount`, which are both immune to forks.

Reply:
I pushed 2 new commits to add the endpoint, I did lot's of manual test with the prod data it works great

Reply:
If one input is joint with 2 outputs, does Postgresql update the row twice?

Reply:
hum good question, as on the first update `AND inputs.output_ref_address IS NULL` won't be true anymore...I would need to try, but if it's updated twice it's not harmful.

Reply:
Sure, it's not a big problem